src/mini_trainer/setup_model_for_training.py

```diff
@@ -174,11 +187,28 @@ def setup_model(
     base_model_args = {
         "pretrained_model_name_or_path": model_name_or_path,
     }
-    # Check if flash_attn is available, otherwise use eager
-    # in practice we will need flash attention when running this repo
+    
+    # Get model config to check for GPT-OSS and set appropriate configurations
+    model_config = AutoConfig.from_pretrained(model_name_or_path)
+    is_gpt_oss = model_config.model_type == "gpt_oss"
```

Comment:
Nit: When you have string constants like these that you use to check against, it's a good practice to have them as global constants like `MODEL_TYPE_GPT_OSS = "gpt_oss"` so that your main logic here (and anywhere else you have only GPT-OSS specific behavior) you can simply do 

```suggestion
    is_gpt_oss = model_config.model_type == MODEL_TYPE_GPT_OSS
```

Reply:
After further review, just move this logic to the `is_gpt_oss` function (no need for a constant).